api/pkg/db/model/model.go

```diff
@@ -85,4 +93,17 @@ type (
 		ResourceID uint
 		Rating     uint `gorm:"not null;default:null"`
 	}
+
+	UserScope struct {
+		UserID  uint
+		ScopeID uint
+	}
+)
+
+type UserType string
+
+// Types of Users
+const (
+	Normal UserType = "normal"
```

Comment:
`Normal` -> `NormalUser`
`Agent` -> `AgentUser`

Reply:
`normal` -> `user`

Reply:
Can we have as `normal` ---> `user` and `agent` --> `admin`

Reply:
`admin`? It will be too generic ..agent would be good because it has a specific task to do.